packages/stores/src/ui-config/manage-liquidity/add-concentrated-liquidity.ts

```diff
@@ -136,28 +137,46 @@ export class ObservableAddConcentratedLiquidityConfig extends TxChainSetter {
     this.fetchHistoricalChartData();
 
     // Calculate quote amount when base amount is input and anchor is base
+    // Calculate an amount0 given an amount1
     autorun(() => {
       const baseAmountRaw =
         this.baseDepositAmountIn.getAmountPrimitive().amount;
-      const baseAmount = new Int(baseAmountRaw);
+      const amount1 = new Int(baseAmountRaw);
       const anchor = this._anchorAsset;
 
-      if (anchor !== "base" || baseAmount.lt(new Int(0))) return;
+      if (anchor !== "base" || amount1.lt(new Int(0))) return;
 
-      if (baseAmount.isZero()) this.quoteDepositAmountIn.setAmount("0");
+      if (amount1.isZero()) this.quoteDepositAmountIn.setAmount("0");
+
+      // Have amount 1
+      // First, determine liquidity0
 
-      // calculate proportional amount of other amount
       const [lowerTick, upperTick] = this.tickRange;
-      const quoteAmount = calculateDepositAmountForQuote(
-        this.pool.currentSqrtPrice,
+      const upperTickSqrt = tickToSqrtPrice(upperTick);
+
+      let sqrtPriceA = this.pool.currentSqrtPrice;
+      let sqrtPriceB = upperTickSqrt;
+
+      if (sqrtPriceA.gt(sqrtPriceB)) {
+        sqrtPriceA = upperTickSqrt;
+        sqrtPriceB = this.pool.currentSqrtPrice;
+      }
+
+      const liquidity0 = amount1
+        .toDec()
+        .mul(sqrtPriceA.mul(sqrtPriceB))
+        .quo(sqrtPriceB.sub(sqrtPriceA));
+
```

Comment:
Could we try to move this liquidity0 (and liquidity1 calc below) into the body of the math functions?

Reply:
Agreed, done here https://github.com/osmosis-labs/osmosis-frontend/pull/1629/commits/b7a16a9266f79f95ae71a5192c7f72eb4fa38ab0